irods/session.py

```diff
@@ -357,6 +357,8 @@ def connection_timeout(self, seconds):
             if seconds < 0 or str(seconds) == 'nan' or str(abs(seconds)) == 'inf':
                 exc = ValueError("The iRODS connection_timeout may not be assigned a negative or otherwise rogue value (eg: NaN, Inf).")
                 raise exc
+        elif seconds is None:
+            pass
         else:
             exc = ValueError("The iRODS connection_timeout must be given a positive value of type int or float.")
```

Comment:
Should we mention `None` as well?

Reply:
In particular, should we mention that `None` is an acceptable value and that it disables socket timeouts?

Reply:
Are you asking about adding a comment here stating that? If so, then a comment in the body of that if-stmt would be good.

Reply:
I see now.

Yes, I agree we need to revise that statement to cover `None`.

Reply:
RIght.  THought I'd done it, but hadn't... will do.

Reply:
> In particular, should we mention that `None` is an acceptable value and that it disables socket timeouts?

Note - `None` is an acceptable value but sets timeout to infinite duration, ie. it disables them.
`0` is not acceptable because socket_obj.settimeout(0) has the effect of setting the socket to nonblocking mode.

Reply:
This is all resolved, I believe. Closing discussion.